src/annotator/test/frame-observer-test.js

```diff
@@ -36,92 +28,98 @@ describe('annotator/frame-observer', () => {
       frameObserver = new FrameObserver(
         container,
         onFrameAdded,
         onFrameRemoved
       );
-      sandbox.stub(console, 'warn');
     });
 
     afterEach(() => {
       container.remove();
       frameObserver.disconnect();
       sandbox.restore();
+      $imports.$restore();
     });
 
     it('triggers onFrameAdded when an annotatable iframe is added', async () => {
       let iframe = createAnnotatableIFrame();
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledWith(onFrameAdded, iframe);
 
       iframe = createAnnotatableIFrame('enable-annotation', 'yes');
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledWith(onFrameAdded, iframe);
 
       iframe = createAnnotatableIFrame('enable-annotation', 'true');
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledWith(onFrameAdded, iframe);
 
       iframe = createAnnotatableIFrame('enable-annotation', '1');
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledWith(onFrameAdded, iframe);
 
       iframe = createAnnotatableIFrame('enable-annotation', 'false'); // the actual value of the attribute is irrelevant
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledWith(onFrameAdded, iframe);
     });
 
     it("doesn't trigger onFrameAdded when non-annotatable iframes are added", async () => {
       createAnnotatableIFrame('dummy-attribute');
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.notCalled(onFrameAdded);
     });
 
     it('removal of the annotatable iframe triggers onFrameRemoved', async () => {
       const iframe = createAnnotatableIFrame();
 
-      await waitForFrameObserver();
+      await delay(0);
       assert.calledOnce(onFrameAdded);
       assert.calledWith(onFrameAdded, iframe);
 
       iframe.remove();
 
-      await waitForFrameObserver();
+      await delay(0);
       assert.calledOnce(onFrameRemoved);
       assert.calledWith(onFrameRemoved, iframe);
     });
 
     it('removal of the `enable-annotation` attribute triggers onFrameRemoved', async () => {
       const iframe = createAnnotatableIFrame();
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledOnce(onFrameAdded);
       assert.calledWith(onFrameAdded, iframe);
 
       iframe.removeAttribute('enable-annotation');
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledOnce(onFrameRemoved);
       assert.calledWith(onFrameRemoved, iframe);
     });
 
     it('changing the `src` attribute triggers onFrameRemoved', async () => {
       const iframe = createAnnotatableIFrame();
-      await waitForFrameObserver();
+      await delay(0);
 
       assert.calledOnce(onFrameAdded);
       assert.calledWith(onFrameAdded, iframe);
 
-      iframe.setAttribute('src', document.location);
+      iframe.setAttribute('src', 'http://localhost:1');
```

Comment:
What is the reason for this change?

Reply:
I had two reasons:

* it should probably be `document.location.href`, but because `document.location` is coerced to a string it gives the same result: `document.location.toString() === document.location.href`

* it is lighter: there is a lot of JS code in that is served in the Karma `document.location` URL, that the browser will load. I just want to exercise the unloading of an iframe, so pointing to some non reachable URL, it just enough.